### ISSUE TITLE
Returns false on valid hex string

### ISSUE THREAD

**Issue by radiovisual** [validate-io/color-hexadecimal#2]
According to the [HTML Color specification](https://html.spec.whatwg.org/#colours), a hexstring without the prepended hash sign `#` is considered false (and returns an error), so this part of your API feels incorrect:

```js
isHexColor('#ffffff');
//=> false
```

Because the hash sign is what makes the string of characters recognizable as hex, I think this module should be a bit more flexible and see colors with and without the hash sign equally:

```js
isHexColor('#ffffff');
//=> true

isHexColor('ffffff');
//=> true
```

Thoughts?

**Comment by kgryte** [comment]
If someone wants to validate a color string prefixed with a `#` symbol, s/he can

``` javascript
var color = '#ffffff';

var bool = isHexColor( color.substring( 1 ) );
// returns true
```

Package behavior was not designed toward exclusively accommodating the HTML specification, or any particular specification for that matter. In the case of the HTML specification, that the specification requires a `#` prefix is accidental and not mandatory in the more general world of six digit hexadecimal colors.

As commented elsewhere, the most general form of a string which is a hexadecimal color is a sequence of six digits: XXXXXX. That a user having such strings would need to accommodate a particular specification which may, or may not, be relevant to them is not warranted; e.g.,

``` javascript
var bool = isHexColor( '#'+'ffffff' );
// returns true
```
And further, if the behavior was based on a less restrictive definition of a hexadecimal color (i.e., `'ffffff'` and `'#ffffff'` would both considered acceptable), then users wanting either one or the other would be out of luck.

As it stands now, the more flexible approach and the one which puts the onus on the user to understand the type of data s/he wants to validate is the approach currently taken.

So, in summary, I am not inclined to support a `#` prefix, when validating such values can be easily done in userland.
